### PR TITLE
LPS-182671 Clean up subclasses of class BaseModelPermissionChecker | part 1

### PR DESCRIPTION
**Pull request by W2jx**
resend, modify class `MBThreadModelResourcePermission`, use `thread id` instead of `message id`, because I found that when adding a thread in the Message Boards in portal, the permission of thread is checked through `thread id`,

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4499

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:document

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
no conflicts(5/22 9:08)

**Comment by W2jx** [comment]
no conflicts(5/23 8:20)

**Comment by W2jx** [comment]
no conflicts(5/24 13:44)

**Comment by W2jx** [comment]
no conflicts(5/25 10:06)

**Comment by W2jx** [comment]
no conflicts(5/26 15:09)

**Comment by W2jx** [comment]
only do semver, no need to rerun test

**Comment by dantewang** [comment]
@yuhai Could you take a look at this one? Thanks!

**Comment by W2jx** [comment]
no conflicts(5/31 9:42)

File: modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLSelectFolderDisplayContext.java [hunk 1]
```diff
@@ -318,6 +321,12 @@ private boolean _isMountFolderVisible() {
 	private static final Log _log = LogFactoryUtil.getLog(
 		DLSelectFolderDisplayContext.class);
 
+	private static final Snapshot<ModelResourcePermission<Folder>>
+		_folderModelResourcePermissionSnapshot = new Snapshot<>(
+			DLSelectFolderDisplayContext.class,
+			Snapshot.cast(ModelResourcePermission.class),
+			"(model.class.name=" + Folder.class.getName() + ")");
+
```
Comment (yuhai) [inline modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLSelectFolderDisplayContext.java:329]: @W2jx, `new DLSelectFolderDisplayContext(...) `only happened in DLFolderItemSelectorView.class and DLFolderItemSelectorView is a @Component, so you may reference this in DLFolderItemSelectorView as the below and then make _dlFolderModelResourcePermission as constructor param for DLSelectFolderDisplayContext used.
```
@Reference(
		target = "(model.class.name=com.liferay.document.library.kernel.model.DLFolder)"
	)
	private ModelResourcePermission<DLFolder> _dlFolderModelResourcePermission;
```

**Pull request closed without merging**